addon/-private/system/store/finders.js

```diff
@@ -52,9 +52,13 @@ export function _find(adapter, store, modelClass, id, internalModel, options) {
   }, `DS: Extract payload of '${modelName}'`);
 }
 
-export function _findMany(adapter, store, modelName, ids, internalModels) {
+export function _findMany(adapter, store, modelName, ids, internalModels, snapshots) {
   if (DEBUG) { incrementRequestCount(); }
-  let snapshots = A(internalModels).invoke('createSnapshot');
+  if (snapshots) {
+    snapshots = A(snapshots);
+  } else {
+    snapshots = A(internalModels).invoke('createSnapshot');
```

Comment:
in what scenario do we reach here without snapshots?

Reply:
I think I've merely added this as a safeguard, so that add-ons relying on this intimate API don't break. 

I'll check if Ember Data itself accesses this from different places, tho. I can also remove this, if you like.